pkg/testutil/prometheus.go

```diff
@@ -100,20 +101,23 @@ func (p *Prometheus) Addr() string {
 }
 
 // SetConfig updates the contents of the config file.
-func (p *Prometheus) SetConfig(s string) error {
+func (p *Prometheus) SetConfig(s string) (err error) {
 	f, err := os.Create(filepath.Join(p.dir, "prometheus.yml"))
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+	defer runutil.BestEffortErr(nil, &err, f, "prometheus config")
```

Comment:
Use LogOnError, I think you care very little if you can't close `prometehus.yml`.